server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -304,6 +306,9 @@ public static NodeMetadata nodeMetadata(Path... dataPaths) throws IOException {
                     } else if (nodeId == null) {
                         nodeId = thisNodeId;
                         version = Version.fromId(Integer.parseInt(userData.get(NODE_VERSION_KEY)));
+                        if (userData.containsKey(OLDEST_INDEX_VERSION_KEY)) {
+                            oldestIndexVersion = Version.fromId(Integer.parseInt(userData.get(OLDEST_INDEX_VERSION_KEY)));
+                        }
```

Comment:
Should we set `oldestIndexVersion` back to `Version.V_EMPTY` here if the key is missing? Otherwise we end up returning a `NodeMetadata` that never really existed. Super-rare case I know...

Reply:
Good catch, it's safest to do it!